app/src/main/java/com/omen273/crossLingo/GameActivity.kt

```diff
@@ -424,14 +445,16 @@ class GameActivity : AppCompatActivity(), CrosswordView.OnLongPressListener,
 
     override fun onCrosswordChanged(view: CrosswordView) {}
 
-    private fun showFinishGameDialog() {
-        val builder = AlertDialog.Builder(this).setMessage(R.string.youve_solved_the_puzzle)
+    private fun showFinishGameDialog(shownAgain: Boolean = false) {
+        val dialog = AlertDialog.Builder(this)
             .setPositiveButton(R.string.reset) { _, _ -> crosswordView.reset() }
             .setNegativeButton(R.string.another_crossword) { _, _ -> onBackPressed() }
             .setNeutralButton(R.string.remove) { _, _ ->
                 delete = true
                 onBackPressed()
-            }.create()
+            }
+        if (!shownAgain) dialog.setMessage(R.string.youve_solved_the_puzzle)
```

Comment:
why do you need this flag?

Reply:
We need it not to show the message R.string.youve_solved_the_puzzle when a user open the crossword again.